src/software/thunderscope/robot_diagnostics/robot_info.py

```diff
@@ -0,0 +1,199 @@
+                                Shows / Hides AI option from control mode menu accordingly
+        :param control_mode_signal: signal that should be emitted when a robot changes control mode
+        """
+
+        super().__init__()
+
+        self.robot_id = robot_id
+        self.control_mode_signal = control_mode_signal
+
+        # when this robot has a battery warning, this is set to True
+        # which prevents spamming the same battery warning
+        # set back to False if battery is back above warning level
+        self.battery_warning_disabled = False
+
+        self.layout = QHBoxLayout()
+
+        self.status_layout = QVBoxLayout()
+
+        # Offsets the minimum of the battery bar from the minimum ideal voltage
+        # Allows battery % to go below the minimum ideal level
+        BATTERY_MIN_OFFSET = 3
+
+        # Battery Bar
+        self.battery_layout = QHBoxLayout()
+        self.battery_progress_bar = common_widgets.ColorProgressBar(
+            MIN_BATTERY_VOLTAGE - BATTERY_MIN_OFFSET, MAX_BATTERY_VOLTAGE
+        )
+        # Battery Voltage Label
+        self.battery_label = QLabel()
+        self.battery_label.setText("NAN")
+        # Label changes when voltage bar level changes
+        self.battery_progress_bar.floatValueChanged.connect(
+            lambda float_val: self.battery_label.setText("%.2fV" % float_val)
+        )
+        self.battery_layout.addWidget(self.battery_progress_bar)
+        self.battery_layout.addWidget(self.battery_label)
+
+        self.status_layout.addLayout(self.battery_layout)
+
+        # Control mode dropdown
+        self.control_mode_layout = QHBoxLayout()
+        self.control_mode_menu = self.create_control_mode_menu(load_fullsystem)
+
+        # Breakbeam status
+        self.breakbeam_label = QLabel()
+        self.breakbeam_label.setText("BREAKBEAM")
+        self.breakbeam_label.setStyleSheet("background-color: grey")
+        self.control_mode_layout.addWidget(self.breakbeam_label)
+        self.control_mode_layout.addWidget(self.control_mode_menu)
+
+        self.status_layout.addLayout(self.control_mode_layout)
+
+        # Vision Pattern
+        self.layout.addWidget(
+            self.create_vision_pattern_label(Colors.ROBOT_MIDDLE_BLUE, ROBOT_RADIUS)
+        )
+
+        self.layout.addLayout(self.status_layout)
+
+        self.setLayout(self.layout)
+
+    def create_control_mode_menu(self, load_fullsystem):
+        """
+        Creates the drop down menu to select the input for each robot
+        :param robot_id: the id of the robot this menu belongs to
+        :param load_fullsystem: whether fullsystem is also an option
+        :return: QComboBox object
+        """
+        control_mode_menu = QComboBox()
+
+        control_mode_menu.addItems(["None", "Manual"])
+        control_mode_menu.setCurrentIndex(0)
+
+        if load_fullsystem:
+            control_mode_menu.addItem("AI")
+            control_mode_menu.setCurrentIndex(2)
```

Comment:
nit:

```suggestion
        control_mode_menu.addItems([IndividualRobotMode.NONE.name, IndividualRobotMode.MANUAL.name])
        control_mode_menu.setCurrentIndex(IndividualRobotMode.NONE)

        if load_fullsystem:
            control_mode_menu.addItem(IndividualRobotMode.AI.name)
            control_mode_menu.setCurrentIndex(IndividualRobotModel.AI)
```

Reply:
ooh interesting didn't know you could do that